test/tests/functional/pbs_ralter.py

```diff
@@ -0,0 +1,1007 @@
+
+    def alter_a_reservation(self, rid, start, end, shift,
+                            alter_s=False, alter_e=False,
+                            ExpectSuccess=1, confirm=True,
+                            interactive=0, sequence=1):
+        """
+        Helper method for altering a reservation.
+        This method also checks for the server and accounting logs.
+
+        param rid: Reservation id.
+        type  rid: string.
+
+        param start: Start time of the reservation.
+        type  start: int.
+
+        param end: End time of the reservation.
+        type  end: int.
+
+        param shift: Time in seconds the reservation times will be moved.
+        type  shift: int.
+
+        param alter_s: Whether the caller intents to change the start time.
+                       Default - No.
+        type  alter_s: bool.
+
+        param alter_e: Whether the caller intents to change the end time.
+                       Default - No.
+        type  alter_e: bool.
+
+        param ExpectSuccess: Whether the caller expects the reservation
+                             to be successful or not.
+                             Default: Yes.
+        type  ExpectSuccess: int.
+
+        param confirm: The expected state of the reservation after it is
+                       altered. It can be either Confirmed or Running.
+                       Default - Confirmed State.
+        type  confirm: bool.
+
+        param interactive: Time in seconds the CLI waits for a reply.
+                           Default - 0 seconds.
+        type  interactive: int.
+
+        param sequence: To check the number of logs corresponding to alter.
+                        Default: 1
+        type  sequence: int.
+        """
+        duration = end - start
+        new_start = start
+        new_end = end
+        flag = 0
+        attrs = {}
+        bu = BatchUtils()
+        standing = 0
+
+        if (alter_s):
+            new_start = start + shift
+            new_start_conv = bu.convert_seconds_to_resvtime(
+                new_start)
+            attrs.update({'reserve_start': new_start_conv})
+
+        if (alter_e):
+            new_end = end + shift
+            new_end_conv = bu.convert_seconds_to_resvtime(new_end)
+            attrs.update({'reserve_end': new_end_conv})
+
+        if (interactive > 0):
+            attrs.update({'interactive': interactive})
+
+        new_duration = new_end - new_start
+        if (ExpectSuccess):
+            msg = ['']
+            acct_msg = ['']
+
+            if (interactive):
+                if (ExpectSuccess == 1):
+                    msg = "pbs_ralter: " + rid + " CONFIRMED"
+                elif (ExpectSuccess == 2):
+                    msg = "pbs_ralter: " + rid + " UNCONFIRMED"
+                else:
+                    msg = "pbs_ralter: " + rid + " DENIED"
+            else:
+                msg = "pbs_ralter: " + rid + " ALTER REQUESTED"
+
+            ret = self.server.alterresv(rid, attrs)
```

Comment:
would you like me to create a PTL ticket for changing alterjob as for my purposes I want the full dictionary being returned from alterresv.